source/droid/gateway/compression.d

```diff
@@ -0,0 +1,37 @@
+module droid.gateway.compression;
+
+import droid.exception;
+import std.zlib,
+       std.stdio,
+       std.conv : to;
+
+enum CompressionType : string {
+    NONE        = "",
+    ZLIB        = "zlib",
+    ZLIB_STREAM = "zlib-stream"
+}
+class Decompressor {
+    string read(ubyte[] data) {
+        throw new DroidException("Compression type not supported!");
+    }
+}
+
+class ZLibStream : Decompressor {
+    const ulong[] ZLIB_SUFFIX = [0x0, 0x0, 0xFF, 0xFF];
```

Comment:
I'm pretty sure `ubyte[]` should be used.

Reply:
Ah right, I must've mistyped.